src/lib/caching/structures/channels/NewsChannel.ts

```diff
@@ -16,6 +16,15 @@ export class NewsChannel extends GuildTextChannel {
 	 */
 	public readonly type = ChannelType.GuildAnnouncement;
 
+	/**
+	 * Crossposts a Message in this channel.
+	 * @param messageID The ID of the {@link Message message} that should be crossposted.
+	 * @since 0.0.1
+	 */
+	public async crosspost(messageID: string): Promise<APIMessageData> {
```

Comment:
This should return a Promise<Message>. See other text channel like methods for converting APIMessageData into patching existing or returning a new Message.

Reply:
To clarify, you'd like the original message to be patched by this method instead of returning the APIMessageData to the Message structure to be patched there